groups/bsl/bslstl/bslstl_ratio.t.cpp

```diff
@@ -0,0 +1,386 @@
+            ASSERT(!(bsl::ratio_greater_equal<NegTwo3rds, Zero>::value));
+            ASSERT( (bsl::ratio_greater_equal<ThreeHalfs, Zero>::value));
+            ASSERT( (bsl::ratio_greater_equal<ThreeHalfs, Six4ths>::value));
+
+            ASSERT(!(bsl::ratio_equal<NegTwo3rds, ThreeHalfs>::value));
+            ASSERT(!(bsl::ratio_equal<NegTwo3rds, Zero>::value));
+            ASSERT(!(bsl::ratio_equal<ThreeHalfs, Zero>::value));
+            ASSERT( (bsl::ratio_equal<ThreeHalfs, Six4ths>::value));
+
+            ASSERT( (bsl::ratio_not_equal<NegTwo3rds, ThreeHalfs>::value));
+            ASSERT( (bsl::ratio_not_equal<NegTwo3rds, Zero>::value));
+            ASSERT( (bsl::ratio_not_equal<ThreeHalfs, Zero>::value));
+            ASSERT(!(bsl::ratio_not_equal<ThreeHalfs, Six4ths>::value));
+
+#if defined(BSLS_LIBRARYFEATURES_HAS_CPP17_BASELINE_LIBRARY) || \
+    defined(BSLS_LIBRARYFEATURES_HAS_CPP14_BASELINE_LIBRARY)
+            // same tests, with using the '_v' suffix
+            ASSERT( (bsl::ratio_less_v<NegTwo3rds, ThreeHalfs>));
+            ASSERT( (bsl::ratio_less_v<NegTwo3rds, Zero>));
+            ASSERT(!(bsl::ratio_less_v<ThreeHalfs, Zero>));
+            ASSERT(!(bsl::ratio_less_v<ThreeHalfs, Six4ths>));
+
+            ASSERT( (bsl::ratio_less_equal_v<NegTwo3rds, ThreeHalfs>));
+            ASSERT( (bsl::ratio_less_equal_v<NegTwo3rds, Zero>));
+            ASSERT(!(bsl::ratio_less_equal_v<ThreeHalfs, Zero>));
+            ASSERT( (bsl::ratio_less_equal_v<ThreeHalfs, Six4ths>));
+
+            ASSERT(!(bsl::ratio_greater_v<NegTwo3rds, ThreeHalfs>));
+            ASSERT(!(bsl::ratio_greater_v<NegTwo3rds, Zero>));
+            ASSERT( (bsl::ratio_greater_v<ThreeHalfs, Zero>));
+            ASSERT(!(bsl::ratio_greater_v<ThreeHalfs, Six4ths>));
+
+            ASSERT(!(bsl::ratio_greater_equal_v<NegTwo3rds, ThreeHalfs>));
+            ASSERT(!(bsl::ratio_greater_equal_v<NegTwo3rds, Zero>));
+            ASSERT( (bsl::ratio_greater_equal_v<ThreeHalfs, Zero>));
+            ASSERT( (bsl::ratio_greater_equal_v<ThreeHalfs, Six4ths>));
+
+            ASSERT(!(bsl::ratio_equal_v<NegTwo3rds, ThreeHalfs>));
+            ASSERT(!(bsl::ratio_equal_v<NegTwo3rds, Zero>));
+            ASSERT(!(bsl::ratio_equal_v<ThreeHalfs, Zero>));
+            ASSERT( (bsl::ratio_equal_v<ThreeHalfs, Six4ths>));
+
+            ASSERT( (bsl::ratio_not_equal_v<NegTwo3rds, ThreeHalfs>));
+            ASSERT( (bsl::ratio_not_equal_v<NegTwo3rds, Zero>));
+            ASSERT( (bsl::ratio_not_equal_v<ThreeHalfs, Zero>));
+            ASSERT(!(bsl::ratio_not_equal_v<ThreeHalfs, Six4ths>));
+#endif
+        }
+
+        if (verbose) printf("\tTest 'ratio' arithmetic operations.\n");
+        {
+            typedef bsl::ratio<3, 2> ThreeHalfs;
+            typedef bsl::ratio<3, 2> Six4ths;
+            typedef bsl::ratio<0, 1> Zero;
+            typedef bsl::ratio<-2, 3> NegTwo3rds;
+
+            ASSERT((bsl::ratio_equal_v<
```

Comment:
This block also uses `_v` stuff, so must also be guarded by the `HAS_CPP17` and `HAS_CPP14` macros as above.

Reply:
I changed these all to use `ratio_equal<...>::value` instead.